boards/st/b_u585i_iot02a/b_u585i_iot02a_stm32u585xx_ext_flash_app.dts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2025 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include "b_u585i_iot02a-common.dtsi"
+#include <zephyr/dt-bindings/memory-attr/memory-attr.h>
+#include <zephyr/dt-bindings/memory-attr/memory-attr-arm.h>
+
+/ {
+	model = "STMicroelectronics B-U585I-IOT02A discovery kit";
+	compatible = "st,b-u585i-iot02a";
+
+	chosen {
+		zephyr,console = &usart1;
+		zephyr,shell-uart = &usart1;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
```

Comment:
What is the purpose of `zephyr,flash` here?

Reply:
Removing  
```
chosen {
		zephyr,flash = &flash0;
}
```

and the samples/hello_world/ cannot build anymore
```
section `rom_start' will not fit in region `FLASH'
 region `FLASH' overflowed by 17616 bytes
```

 It should though without any  mcuboot